### PR TITLE
ZIO 2.0.0-RC1 upgrade

### PR DESCRIPTION
**Pull request by yisraelU**
upgraded to RC1 of ZIO 2.0.0
Added AMQPConfig class with default config and new connection method

### CODE REVIEW

**Comment by svroonland** [comment]
Nice!

I made a fix for the failing build in this commit, perhaps you could cherry-pick it:  `c57c271883c277632121a508f380f47878cbfc12`. I can't push to your master branch, but I believe if you make a PR from a branch on your fork, I can push changes to it as well.

File: src/main/scala/nl/vroste/zio/amqp/model/AMQPConfig.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package nl.vroste.zio.amqp.model
+import java.net.URI
+
+case class AMQPConfig(
+  user: String,
+  password: String,
+  vhost: String,
+  heartbeatInterval: Short,
```
Comment (svroonland) [inline src/main/scala/nl/vroste/zio/amqp/model/AMQPConfig.scala:8]: Shall we make this a zio Duration so we know what the unit is (now I'm not sure if it's seconds or milliseconds)

File: src/main/scala/nl/vroste/zio/amqp/model/AMQPConfig.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  ssl: Boolean,
+  host: String,
+  port: Short,
+  connectionTimeout: Short
```
Comment (svroonland) [inline src/main/scala/nl/vroste/zio/amqp/model/AMQPConfig.scala:12]: Shall we make this a zio Duration?

File: project/Dependencies.scala [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
 
 object Dependencies {
-  val zioVersion = "2.0.0-M4"
+
+  val zioVersion = "2.0.0-RC1"
   lazy val deps  = Seq(
     "dev.zio"                %% "zio-streams"                 % zioVersion,
     "dev.zio"                %% "zio-test"                    % zioVersion % Test,
     "dev.zio"                %% "zio-test-sbt"                % zioVersion % Test,
+    "dev.zio"                %% "zio-config"                  % "1.0.10",
```
Comment (svroonland) [inline project/Dependencies.scala:10]: I don't think this is used

**Review by svroonland — CHANGES_REQUESTED** [review]
Looking good, a few remarks

File: project/Dependencies.scala (hunk 3 above)
Comment (yisraelU) [reply project/Dependencies.scala:10]: True I was planning on using it for other code for loading config , just didn't want to push that info this commit .
I'll remove it

File: src/main/scala/nl/vroste/zio/amqp/model/AMQPConfig.scala (hunk 1 above)
Comment (yisraelU) [reply src/main/scala/nl/vroste/zio/amqp/model/AMQPConfig.scala:8]: Yes , I'll do that for all time related fields

**Comment by yisraelU** [comment]
> Nice!
> 
> I made a fix for the failing build in this commit, perhaps you could cherry-pick it: `c57c271883c277632121a508f380f47878cbfc12`. I can't push to your master branch, but I believe if you make a PR from a branch on your fork, I can push changes to it as well.
@svroonland 
Seems like unused imports cause failure. I removed them.
The test thats failing now needs to be changed as it runs the wrong version of Scala 3. which is incompatible with ZIO 2.0.0 RC1 . in the build.sbt I switched from 3.0.1 to 3.1.0 , however the tests are hardcoded to 3.0.1

**Comment by svroonland** [comment]
Yeah for that we need to make the changes to the config.yml file like in the commit I referenced.

**Comment by yisraelU** [comment]
> Yeah for that we need to make the changes to the config.yml file like in the commit I referenced.

@svroonland sorry , I missed that.
All done

**Review by svroonland — APPROVED** [review]

**Comment by svroonland** [comment]
Great, thanks! I merged it, it should be in the snapshot soon.

**Pull request merged**